documentation/docs/developers/connectors/task4.md

```diff
@@ -1,8 +1,223 @@
+      RateLimit(limit_id=CANCEL_ORDER_PATH_URL, limit=15, time_interval=0.1),
+      RateLimit(limit_id=GET_ACCOUNT_SUMMARY_PATH_URL, limit=3, time_interval=0.1),
+      RateLimit(limit_id=GET_ORDER_DETAIL_PATH_URL, limit=30, time_interval=0.1),
+      RateLimit(limit_id=GET_OPEN_ORDERS_PATH_URL, limit=3, time_interval=0.1),
+  ]
+  ```
+
+!!! note
+    `time_interval` here is in seconds. i.e. The rate limits for `CREATE_ORDER_PATH_URL` is 15 requests every 100ms
+
+#### 2. Rate Limit Pools
+
+Rate limit pools refer to a group of endpoints that consumes from a single rate limit. For this rate limit type, the key information to retrieve for each endpoint are its assigned pool(s) and its respective limit and time interval.
+
+!!! note
+    Examples of existing connectors that utilizes this rate limit implementation are:</br>
+      (1) Binance, Binance Perpetual</br>
+      (2) Ndax</br>
+
+##### Configuring Rate Limits
+
+An example of an exchange implementing this can be seen in the Ndax connector.
+
+Note
+
+  All the rate limit are initialized in the `ndax_constants.py` file. 
+
+  ```python
+  # Pool IDs
+  HTTP_ENDPOINTS_LIMIT_ID = "AllHTTP"
+  WS_ENDPOINTS_LIMIT_ID = "AllWs"
+
+  RATE_LIMITS = [
+    # REST API Pool(applies to all REST API endpoints)
+    RateLimit(limit_id=HTTP_ENDPOINTS_LIMIT_ID, limit=HTTP_LIMIT, time_interval=MINUTE),
+    # WebSocket Pool(applies to all WS requests)
+    RateLimit(limit_id=WS_ENDPOINTS_LIMIT_ID, limit=WS_LIMIT, time_interval=MINUTE),
+    # Public REST API endpoint
+    RateLimit(
+        limit_id=MARKETS_URL,
+        limit=HTTP_LIMIT,
+        time_interval=MINUTE,
+        linked_limits=[LinkedLimitWeightPair(HTTP_ENDPOINTS_LIMIT_ID)],
+    ),
+    # WebSocket Auth endpoint
+    RateLimit(
+        limit_id=ACCOUNT_POSITION_EVENT_ENDPOINT_NAME,
+        limit=WS_LIMIT,
+        time_interval=MINUTE,
+        linked_limits=[LinkedLimitWeightPair(WS_ENDPOINTS_LIMIT_ID)],
+    ),
+  ]
+  ```
+
+!!! note
+    Notice that we assign an abitruary limit id(i.e. `HTTP_ENDPOINTS_LIMIT_ID`) to the API pools and we use the [`LinkedLimitWeightPair`](https://github.com/CoinAlpha/hummingbot/blob/master/hummingbot/core/api_throttler/data_types.py) to assign an endpoint to the API pool. Also do note that an endpoint may belong to multiple endpoints. It is also worth noting that there can be more complex implementations to API pools as seen in the ByBit Perpetual connector [here](https://github.com/CoinAlpha/hummingbot/blob/master/hummingbot/connector/derivative/bybit_perpetual/bybit_perpetual_constants.py).
+
+#### 3. Weighted Request Rate Limits
+
+For weighted rate limits, each endpoint is assigned a request weight. Generally, these exchanges would utilize Rate Limit Pools in conjunction with the request weights, where different endpoints will have a different impact on the given pool. Key information to retrieve for these exchanges are the weights for each endpoint, limits the time interval for the API Pool.
```

Comment:
I don't understand the last part of the sentence, after the last comma: `..., limits the time interval for the API Pool.`

Reply:
its should be "..., limits **and** the time internals for the API Pool."